tests/rptest/services/redpanda.py

```diff
@@ -352,10 +352,18 @@ class SecurityConfig:
 
     def __init__(self):
         self.enable_sasl = False
+        self.kafka_enable_authorization: Optional[bool] = None
+        self.endpoint_authn_method: Optional[str] = None
         self.tls_provider: Optional[TLSProvider] = None
 
-        # extract principal from mtls distinguished name
-        self.enable_mtls_identity = False
+    # sasl is required
+    def sasl_enabled(self):
+        return (self.kafka_enable_authorization is None
+                and self.enable_sasl) or self.endpoint_authn_method == "sasl"
```

Comment:
I'm still confused about the meaning of `kafka_enable_authorization`.  like i could have it set to False and then authn_method == sasl and sasl is still enabled?